code/modules/recycling/sortingmachinery.dm

```diff
@@ -235,101 +255,173 @@
 
 	var/c_mode = 0
 
-	New()
-		..()
-		spawn(5)
-			trunk = locate() in src.loc
-			if(trunk)
-				trunk.linked = src	// link the pipe trunk to self
+/obj/machinery/disposal/deliveryChute/New()
+	..()
+	spawn(5)
+		trunk = locate() in src.loc
+		if(trunk)
+			trunk.linked = src	// link the pipe trunk to self
 
-	interact()
-		return
+/obj/machinery/disposal/deliveryChute/interact()
+	return
 
-	update()
-		return
+/obj/machinery/disposal/deliveryChute/update()
+	return
 
-	Bumped(var/atom/movable/AM) //Go straight into the chute
-		if(istype(AM, /obj/item/projectile))  return
-		switch(dir)
-			if(NORTH)
-				if(AM.loc.y != src.loc.y+1) return
-			if(EAST)
-				if(AM.loc.x != src.loc.x+1) return
-			if(SOUTH)
-				if(AM.loc.y != src.loc.y-1) return
-			if(WEST)
-				if(AM.loc.x != src.loc.x-1) return
-
-		if(istype(AM, /obj))
-			var/obj/O = AM
-			O.loc = src
-		else if(istype(AM, /mob))
-			var/mob/M = AM
-			M.loc = src
-		src.flush()
-
-	flush()
-		flushing = 1
-		flick("intake-closing", src)
-		var/deliveryCheck = 0
-		var/obj/structure/disposalholder/H = new()	// virtual holder object which actually
+/obj/machinery/disposal/deliveryChute/Bumped(atom/movable/AM) //Go straight into the chute
+	if(istype(AM, /obj/item/projectile))  return
+	switch(dir)
+		if(NORTH)
+			if(AM.loc.y != src.loc.y+1) return
+		if(EAST)
+			if(AM.loc.x != src.loc.x+1) return
+		if(SOUTH)
+			if(AM.loc.y != src.loc.y-1) return
+		if(WEST)
+			if(AM.loc.x != src.loc.x-1) return
+
+	if(istype(AM, /obj))
+		var/obj/O = AM
+		O.loc = src
+	else if(istype(AM, /mob))
+		var/mob/M = AM
+		M.loc = src
+	src.flush()
+
+/obj/machinery/disposal/deliveryChute/flush()
+	flushing = 1
+	flick("intake-closing", src)
+	var/deliveryCheck = 0
+	var/obj/structure/disposalholder/H = new()	// virtual holder object which actually
 													// travels through the pipes.
-		for(var/obj/structure/bigDelivery/O in src)
-			deliveryCheck = 1
-			if(O.sortTag == 0)
-				O.sortTag = 1
-		for(var/obj/item/smallDelivery/O in src)
-			deliveryCheck = 1
-			if(O.sortTag == 0)
-				O.sortTag = 1
-		if(deliveryCheck == 0)
-			H.destinationTag = 1
-
-		sleep(10)
-		playsound(src, 'sound/machines/disposalflush.ogg', 50, 0, 0)
-		sleep(5) // wait for animation to finish
-
-		H.init(src)	// copy the contents of disposer to holder
-		air_contents = new() // The holder just took our gas; replace it
-		H.start(src) // start the holder processing movement
-		flushing = 0
-		// now reset disposal state
-		flush = 0
-		if(mode == 2)	// if was ready,
-			mode = 1	// switch to charging
-		update()
+	for(var/obj/structure/bigDelivery/O in src)
+		deliveryCheck = 1
+		if(O.sortTag == 0)
+			O.sortTag = 1
+	for(var/obj/item/smallDelivery/O in src)
+		deliveryCheck = 1
+		if(O.sortTag == 0)
+			O.sortTag = 1
+	for(var/obj/item/shippingPackage/O in src)

```

Comment:
This section of logic handling is new, since it is the new shippingPackage handling. It is slightly hidden by the removing of relative pathing around this addition.

This is the only bit of new code in the sections where I removed relative pathing.